README.md

```diff
@@ -142,42 +142,60 @@ Once the libraries are added to a project, the basics of adding and reading
 
-Datastore ds = manager.openDatastore("my_datastore");
-
-// Create a document
-DocumentRevision revision = new DocumentRevision();
-Map<String, Object> body = new HashMap<String, Object>();
-body.put("animal", "cat");
-revision.setBody(DocumentBodyFactory.create(body));
-DocumentRevision saved = ds.createDocumentFromRevision(revision);
-
-// Add an attachment -- binary data like a JPEG
-UnsavedFileAttachment att1 = new UnsavedFileAttachment(new File("/path/to/image.jpg"),
-                                                       "image/jpeg");
-saved.getAttachments().put(att1.name, att1);
-DocumentRevision updated = ds.updateDocumentFromRevision(saved);
-
-// Read a document
-DocumentRevision aRevision = ds.getDocument(updated.getId());
+try {
+    Datastore ds = manager.openDatastore("my_datastore");
+
+    // Create a document
+    DocumentRevision revision = new DocumentRevision();
+    Map<String, Object> body = new HashMap<String, Object>();
+    body.put("animal", "cat");
+    revision.setBody(DocumentBodyFactory.create(body));
+    DocumentRevision saved = ds.createDocumentFromRevision(revision);
+
+    // Add an attachment -- binary data like a JPEG
+    UnsavedFileAttachment att1 =
+        new UnsavedFileAttachment(new File("/path/to/image.jpg"), "image/jpeg");
+    saved.getAttachments().put(att1.name, att1);
+    DocumentRevision updated = ds.updateDocumentFromRevision(saved);
+
+    // Read a document
+    DocumentRevision aRevision = ds.getDocument(updated.getId());
+} catch (DatastoreException datastoreException) {
+
+    // this will be thrown if we don't have permissions to write to the
+    // datastore path
+    System.err.println("Problem opening datastore: "+datastoreException);
+} catch (DocumentException documentException) {
+
+    // this will be thrown in case of errors performing CRUD operations on
+    // documents
+    System.err.println("Problem accessing datastore: "+documentException);
+}
 ```
 
 Read more in [the CRUD document](https://github.com/cloudant/sync-android/blob/master/doc/crud.md).
 
 You can also subscribe for notifications of changes in the database, which
 is described in [the events documentation](https://github.com/cloudant/sync-android/blob/master/doc/events.md).
 
+The
+[javadoc](http://www.javadoc.io/doc/com.cloudant/cloudant-sync-datastore-core/)
+for the latest release version of the library is the definitive
+reference for the library. Each jar contains the full javadoc for
+itself and the other jars - this is for convenience at the slight
+expense of duplication.
+
+Note that each class has an "API Status" declaration in the
+javadoc. This status is either "Public" or "Private". The general
+guidance is that API consumers (who are building software which uses
+this library) are discouraged from using "Private" API classes. This
+is because they may expose implementation details of the library which
+may be subject to change without notice.
```

Comment:
As the first thing that people see, it should be in README, there is a chance some people won't read the change log.